README.md

```diff
@@ -1,2 +1,437 @@
+      {'Ipv6Address': 'eb7a:5a31:f899:dd8c:e566:3307:a45e:dcf6'}],
+     'Ipv6AddressCount': 3,
+     'PrivateIpAddress': '80.141.152.14'}]},
+  'CpuOptions': {'CoreCount': 4, 'ThreadsPerCore': 1},
+  'CreditSpecification': {'CpuCredits': 'unlimited'},
+  'LaunchTemplateId': 'lt-aaa68831cce2a8d91',
+  'VersionNumber': 4},
+ {'CreateTime': datetime.datetime(2017, 11, 20, 19, 4, 54),
+  'DefaultVersion': False,
+  'LaunchTemplateData': {'ImageId': 'ami-2f7ac02a',
+   'KeyName': 'kp-us-east',
+   'NetworkInterfaces': [{'DeviceIndex': 0,
+     'Groups': ['sg-1c628b25'],
+     'SubnetId': 'subnet-a4579fe6',
+     'Ipv6Addresses': [{'Ipv6Address': 'f486:915c:2be9:b0da:7d60:3fae:d65a:e8d8'},
+      {'Ipv6Address': 'eb7a:5a31:f899:dd8c:e566:3307:a45e:dcf6'}],
+     'Ipv6AddressCount': 2,
+     'PrivateIpAddress': '80.141.152.136'}]},
+  'CpuOptions': {'CoreCount': 3, 'ThreadsPerCore': 2},
+  'CreditSpecification': {'CpuCredits': 'standard'},
+  'LaunchTemplateId': 'lt-aaa68831cce2a8d91',
+  'VersionNumber': 5}]
+```
+
+#### Filtering by object attributes
+
+This can be useful if you're working with objects that have a lot
+of "has-a" relationships to other objects. For brevity,
+a hacky binary tree-like class is used to build a fictional ancestor chart.
+
+```python
+>>> class Node:
+    instances = []
+    def __init__(self, name, mother=None, father=None):
+        self.name = name
+        self.mother = mother
+        self.father = father
+        self.instances.append(self)
+    def __repr__(self):
+        return (f"Node('{self.name}', mother={repr(self.mother)}, "
+                f"father={repr(self.father)})")
+>>> Node(name='Tiya Meadows',
+         mother=Node('Isobel Meadows (nee Walsh)',
+                     mother=Node(name='Laura Walsh (nee Stanton)',
+                                 mother=Node('Opal Eastwood (nee Plant)'),
+                                 father=Node('Alan Eastwood')),
+                     father=Node(name='Jimmy Walsh')),
+         father=Node(name='Isaac Meadows',
+                     mother=Node('Halle Meadows (nee Perkins)'),
+                     father=Node('Wilbur Meadows')))
+```
+
+To demonstrate the syntax, we can filter for the root node by their
+great-great-grandmother.
+
+```python
+>>> results = q_filter(
+        Node.instances,
+        q_attr("mother.mother.mother.name").contains("Opal Eastwood")
+    )
+>>> list(results)
+[Node('Tiya Meadows', mother=Node('Isobel Meadows (nee Walsh)', mother=Node('Laura Walsh (nee Stanton)', mother=Node('Opal Eastwood (nee Plant)', mother=None, father=None), father=Node('Alan Eastwood', mother=None, father=None)), father=Node('Jimmy Walsh', mother=None, father=None)), father=Node('Isaac Meadows', mother=Node('Halle Meadows (nee Perkins)', mother=None, father=None), father=Node('Wilbur Meadows', mother=None, father=None)))]
+```
+
+As attributes names can only be alphabetic characters and underscores,
```

Comment:
Technically, this is not true:
```py
>>> class C(): pass
... 
>>> c = C()
>>> setattr(c, 'a.b', 123)
>>> getattr(c, 'a.b')
123
>>> c.__dict__
{'a.b': 123}
```

Reply:
That's interesting. I tried using calling `setattr` with a `frozenset`  attribute name, and at least they restrict it to be a string. However the following dunder-headed naughtiness workers:
```c.__dict__[frozenset([1,2,3])] = 123```

 I'll change it to "user-friendly attributes names" or somthing.